hypothesis-python/src/_hypothesis_pytestplugin.py

```diff
@@ -134,13 +151,24 @@ def pytest_configure(config):
             except ValueError:
                 pass
             core.global_force_seed = seed
-        config.addinivalue_line("markers", "hypothesis: Tests which use hypothesis.")
 
     @pytest.hookimpl(hookwrapper=True)
     def pytest_runtest_call(item):
-        if not hasattr(item, "obj"):
+        if not (hasattr(item, "obj") and "hypothesis" in sys.modules):
             yield
-        elif not is_hypothesis_test(item.obj):
+            return
+
+        from hypothesis import HealthCheck, core, settings
+        from hypothesis.errors import InvalidArgument
+        from hypothesis.internal.detection import is_hypothesis_test
+        from hypothesis.internal.escalation import current_pytest_item
+        from hypothesis.internal.healthcheck import fail_health_check
+        from hypothesis.reporting import with_reporter
+        from hypothesis.statistics import collector, describe_statistics
```

Comment:
Is this a performance hazard?

I vaguely recall from https://github.com/HypothesisWorks/hypothesis/pull/1271 that repeated local imports can lead to trouble, though I'm not sure whether the same circumstances apply.

Reply:
Only minimal - the perf killer was uncached *failed* imports, which hit disk every time.  Successful imports are ~8x an empty function call, which is pretty fast, but I can still get some small wins from tighter scoping.